geometry/instant_output.cpp

```diff
@@ -0,0 +1,49 @@
+﻿#include <limits>
+#include <string_view>
+
+#include "absl/strings/str_format.h"
+#include "astronomy/time_scales.hpp"
+#include "geometry/named_quantities.hpp"
+
+namespace principia {
+namespace geometry {
+namespace internal_point {
+
+using astronomy::DateTimeAsTT;
+using astronomy::J2000;
+using astronomy::operator""_TT;
+using astronomy::TTSecond;
+using quantities::Time;
+using quantities::si::Second;
+
+std::ostream& operator<<(std::ostream& os, Instant const& t) {
+  Time const from_j2000 = t - J2000;
+  // Dates before JD0.5 and after the year 9999 are not supported; Sterbenz’s
+  // lemma fails to apply in the second before J2000, we need to print the
+  // sign of 0 for J2000 itself, and the second after J2000 has fractions of a
+  // second that are too small to be reasonably printed in fixed format.
+  if (t >= "JD0.5"_TT && t < "9999-12-31T24:00:00"_TT &&
+      !(t > J2000 - 1 * Second && t < J2000 + 1 * Second)) {
+    auto const tt_second = TTSecond(t);
+    Instant const start_of_second = DateTimeAsTT(tt_second);
+    // This subtraction is exact by Sterbenz’s lemma.
+    Time const remainder = t - start_of_second;
+    // |remainder| being the result of a subtraction of numbers greater than or
+    // equal to 1, it is a multiple of 2u ≈ 2⏨-16; 16 fractional decimal digits
```

Comment:
I have no idea what `2⏨-16` means.

Reply:
https://en.wikipedia.org/wiki/Scientific_notation#:~:text=Decimal%20Exponent%20Symbol%20is%20part,Algol%2060%20and%20Algol%2068.

Reply:
Could we please use a comprehensible notation and not some prehistoric convention? `2E-16`, `2.0E-16`, `2×10⁻¹⁶`.